tfx/experimental/templates/taxi/README.md

```diff
@@ -24,3 +24,63 @@ about the dataset in [Google BigQuery](https://cloud.google.com/bigquery/).
 Explore the full dataset in the
 [BigQuery UI](
 https://bigquery.cloud.google.com/dataset/bigquery-public-data:chicago_taxi_trips).
+
+
+## Testing the pipeline using Stub Executors
+
+### Introduction
+**You should complete [template.ipynb](https://github.com/tensorflow/tfx/blob/master/docs/tutorials/tfx/template.ipynb) tutorial up to *Step 6* in order to proceed this tutorial.**
+
+This document will provide instructions to test a TensorFlow Extended (TFX) pipeline
+using `BaseStubExecuctor`, which generates fake data using the golden test data. This is intended for users to replace executors they don't want to test so that they could save time from running the actual executors. Stub executor is provided with TFX Python package under `tfx.experimental.pipeline_testing.base_stub_executor`.
+
+This tutorial serves as an extension to `template.ipynb` tutorial, thus you will also use [Taxi Trips dataset](
+https://data.cityofchicago.org/Transportation/Taxi-Trips/wrvz-psew)
+released by the City of Chicago. We strongly encourage you to try modifying the components prior to utilizing stub executors.
+
+### Step 1. Record the pipeline outputs in Google Cloud Storage
+
+Since this tutorial assumes that you have completed `template.ipynb` up to step 6, a successful pipeline run must have been saved in the MLMD, which can be accessed using gRPC server. 
+
+Open a Terminal and run the following commands:
+
+> `$ gcloud container clusters get-credentials <cluster_name> --zone <compute_zone> --project {GOOGLE_CLOUD_PROJECT}`
+
+> `$ nohup kubectl port-forward deployment/metadata-grpc-deployment -n <namespace> {PORT}:8080 &`
+
+Open another terminal and run the following command:
+> `$
+python tfx/experimental/pipeline_testing/pipeline_recorder.py \
```

Comment:
One problem is that users will not have tfx source files here without `git clone` or equivalent. Should we include an instruction for cloning or should give an instruction to find and use the installed tfx library?